app/backend/chat_history/cosmosdb.py

```diff
@@ -34,23 +34,50 @@ async def post_chat_history(auth_claims: Dict[str, Any]):
 
     try:
         request_json = await request.get_json()
-        id = request_json.get("id")
-        answers = request_json.get("answers")
-        title = answers[0][0][:50] + "..." if len(answers[0][0]) > 50 else answers[0][0]
+        session_id = request_json.get("id")
+        message_pairs = request_json.get("answers")
+        first_question = message_pairs[0][0]
+        title = first_question + "..." if len(first_question) > 50 else first_question
         timestamp = int(time.time() * 1000)
 
-        await container.upsert_item(
-            {"id": id, "entra_oid": entra_oid, "title": title, "answers": answers, "timestamp": timestamp}
-        )
+        # Insert the session item:
+        session_item = {
+            "id": session_id,
+            "session_id": session_id,
+            "entra_oid": entra_oid,
+            "type": "session",
+            "title": title,
+            "timestamp": timestamp,
+        }
+
+        message_pair_items = []
+        # Now insert a message item for each question/response pair:
+        for ind, message_pair in enumerate(message_pairs):
+            message_pair_items.append(
+                {
+                    "id": f"{session_id}-{ind}",
```

Comment:
Rather than formatting this as a string - should we store the message ordinal as a separate field to enable sorting by history?

Reply:
I think I need an ID regardless, for CosmosDB, right? So that was my attempt at a unique ID.
But yes I can also add as a separate field. @markjbrown says that the default sorting is based on insertion order, but I like the clarity of having a sequence marker too. Note that it'd be null for "session" types.